test/unit/kubernetes-deploy/kubernetes_resource/deployment_test.rb

```diff
@@ -147,6 +175,17 @@ def test_validation_fails_with_invalid_rollout_annotation
     assert_equal expected, deploy.validation_error_msg
   end
 
+  def test_validation_with_percent_rollout_annotation
+    deploy = build_synced_deployment(template: build_deployment_template(rollout: '10%'), replica_sets: [])
+    deploy.kubectl.expects(:run).with('create', '-f', anything, '--dry-run', '--output=name', anything).returns(
+      ["", "super failed", SystemExit.new(1)]
```

Comment:
If we're trying to prove that this value is valid, why not make this command return 0 and `assert deploy_validate_definition`/`assert_empty deploy.validation_error_msg`?